service/controller/resource/monitoring/prometheus/node_count.go

```diff
@@ -0,0 +1,86 @@
+package prometheus
+
+import (
+	"github.com/giantswarm/k8sclient/v7/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"golang.org/x/net/context"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	expv1beta1 "sigs.k8s.io/cluster-api/exp/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/prometheus-meta-operator/v2/service/key"
+)
+
+func getClusterNodeCount(ctx context.Context, k8sClient k8sclient.Interface, cluster metav1.Object) (int, error) {
+	mdCount, err := getClusterNodeCountFromMachineDeployment(ctx, k8sClient, cluster)
+	if err != nil {
+		return 0, microerror.Mask(err)
+	}
+
+	mpCount, err := getClusterNodeCountFromMachinePool(ctx, k8sClient, cluster)
+	if err != nil {
+		return 0, microerror.Mask(err)
+	}
+
+	var nodeCount int32 = 1
+	if mdCount > nodeCount {
+		nodeCount = mdCount
+	}
+	if mpCount > nodeCount {
+		nodeCount = mpCount
+	}
+
+	// TODO: Is this ok ? Are we loosing precision here going from int32 to int ?
```

Comment:
I don't think we lose any kind of precision with integers :)

Reply:
Just a self reminder to check int32 vs int precision

Reply:
That's alright
![image](https://github.com/giantswarm/prometheus-meta-operator/assets/6536819/5d05f3cf-7aed-494e-be14-39afa6fb49be)
https://pkg.go.dev/builtin#int